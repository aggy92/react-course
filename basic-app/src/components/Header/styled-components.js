import styled from 'styled-components'

export const HeaderWrapper = styled.h1`
        font-size: 40px;
        text-align: center;
    `;