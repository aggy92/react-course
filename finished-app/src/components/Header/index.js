import React from 'react';
import { useLocation } from 'react-router-dom';
import { HeaderWrapper } from './Header.css.js';

const Header = () => {
  const { pathname } = useLocation();
  const title = pathname.includes('pokemon') ? 'Pokemon' : 'Pokedex';
  return <HeaderWrapper>{title}</HeaderWrapper>;
};

export default Header;
