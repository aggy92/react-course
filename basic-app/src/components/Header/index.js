import React from 'react';
import {useLocation} from "react-router-dom";
import {HeaderWrapper} from "./styled-components";

export const Header = () => {
    const {pathname} = useLocation();
    console.log(pathname);
    const title = pathname && pathname.includes("pokemon") ? "Details" : "List";
    return <HeaderWrapper>{title}</HeaderWrapper>
};