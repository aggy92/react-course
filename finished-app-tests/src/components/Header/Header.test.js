import React from "react";
import { shallow } from "enzyme";
import Header from "./index";

describe("<Header />", () => {
  it("renders a h1", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.length).toBe(1);
    expect(wrapper.name()).toBe("styled.h1");
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders a title when passed such a prop", () => {
    const wrapper = shallow(<Header title="Pokedeck" />);
    expect(wrapper.text()).toEqual("Test");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
