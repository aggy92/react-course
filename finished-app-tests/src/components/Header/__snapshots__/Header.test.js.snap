// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders a h1 1`] = `"<h1 class=\\"sc-AykKC gzshXR\\"></h1>"`;

exports[`<Header /> renders a title when passed such a prop 1`] = `"<h1 class=\\"sc-AykKC gzshXR\\">Pokedeck</h1>"`;
